fabric-convention-tags-v1/src/main/java/net/fabricmc/fabric/api/tag/convention/v1/TagUtil.java

```diff
@@ -28,11 +31,32 @@
  */
 public class TagUtil {
 	/**
+	* @return if the entry is in the provided tag.
+	*/
+	public static <T> boolean isIn(TagKey<T> tagKey, T entry) {
+		return isIn(null, tagKey, entry);
+	}
+
+	/**
+	 * @param registryManager the registry manager instance of the client or server. If the tag refers to entries
+	 *                        within a dynamic registry, such as {@link net.minecraft.world.biome.Biome}s,
+	 *                        this must be passed to correctly evaluate the tag. Otherwise, the registry is found by
+	 *                        looking in {@link Registry#REGISTRIES}.
 	 * @return if the entry is in the provided tag.
 	 */
 	@SuppressWarnings("unchecked")
-	public static <T> boolean isIn(TagKey<T> tagKey, T entry) {
-		Optional<? extends Registry<?>> maybeRegistry = Registry.REGISTRIES.getOrEmpty(tagKey.registry().getValue());
+	public static <T> boolean isIn(@Nullable DynamicRegistryManager registryManager, TagKey<T> tagKey, T entry) {
+		Optional<? extends Registry<?>> maybeRegistry;
+
+		if (registryManager != null) {
+			maybeRegistry = registryManager.getOptional(tagKey.registry());
+
+			if (maybeRegistry.isEmpty()) {
```

Comment:
I believe this is redundant.

Reply:
The first one pulls from the dynamic registry, the latter pulls from the static registry. I don't believe the dynamic registry contained the static ones?

Reply:
getOptional does fall back to static ones. getOptionalManaged is the one that does not.

Reply:
that it does, fixed